### ISSUE TITLE
Add CLFLUSHOPT support to drmemtrace

### ISSUE THREAD

**Issue by derekbruening** [DynamoRIO/dynamorio#7111]
My pending comment in https://github.com/DynamoRIO/dynamorio/pull/7109 which adds CLFLUSHOPT opcode decoding support:

drmemtrace needs to be udpated as well: it looks like CLFLUSHOPT also invalidates all cache levels and so should be treated like CLFLUSH in memtraces (or would a simulator want to know about the subtle differences?). So just adding to 2 spots in instru.cpp https://github.com/DynamoRIO/dynamorio/blob/master/clients/drcachesim/tracer/instru.cpp#L227 and https://github.com/DynamoRIO/dynamorio/blob/master/clients/drcachesim/tracer/instru.cpp#L244 should be enough (optionally extend invariant_checker_test).  Hmm.  Maybe enough questions this should be separated out.

**Comment by derekbruening** [comment]
We may also want to mark other cache ops such as CLWB and CLDEMOTE.

**Comment by khuey** [comment]
> In fact, [e24dbb0](https://github.com/DynamoRIO/dynamorio/commit/e24dbb0c216b51eef7ab97bad5130e5a74683fae) has CLWB marked as a load: which it is not. If we do nothing, the trace will have it labeled as a read==load record. That highlights that this affects all analyzers and not just those that want to model cache behavior: those looking at loads will do the wrong thing. So this is not just about augmenting the trace but correcting the trace.

fwiw I have a similar issue. Our use of DR is confined to the decoding/encoding components. We examine blocks of instructions and capture their memory inputs so that the block can later be re-executed (with appropriate rewriting) outside of the original program by providing the initial register states and said memory inputs. We end up special casing the prefetch/cache flush instructions to avoid unnecessarily capturing their input memory operands precisely because they are not architectural loads.

**Comment by derekbruening** [comment]
Should there be an IR API routine to query whether a load actually issues a regular load?  A different version of `opc_is_not_a_real_memory_load()`; and maybe export that one too.

**Comment by khuey** [comment]
Yeah that might be nice. I looked at adding an equivalent of `instr_is_prefetch()` for these cache flushing instructions but it looked non-trivial for ARM because there's not a separate opcode there.

**Comment by derekbruening** [comment]
> Yeah that might be nice. I looked at adding an equivalent of `instr_is_prefetch()` for these cache flushing instructions but it looked non-trivial for ARM because there's not a separate opcode there.

Some of us have lobbied to separate the DR versions of some Arm opcodes to make these things easier.  See the discussion at
https://github.com/DynamoRIO/dynamorio/pull/4386#discussion_r462356954 on
opcode philosophy of separate opcodes for separate semantics; also xref #4388 on making drmemtrace's Arm prefetch operation dispatch cleaner.
